### PR TITLE
release-small: prune more obsolete software

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change

This list feels like it was composed in 2007, but besides that.

Went through and pruned packages which filled 2 or more of the following conditions:

- Never heard of it
- Obsolete by a better tool (e.g. `enscript`)
- No longer relevant for nixos (e.g. sysvinit)
- Consistently broken in nixpkgs (e.g. mono)
- Large build closure (e.g. wireshark + qt)

related: #157869 

cc @grahamc @edolstra 

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/release-small.nix [hunk 1]
```diff
@@ -33,20 +33,17 @@ with import ./release-lib.nix { inherit supportedSystems nixpkgsArgs; };
   classpath = linux;
   cmake = all;
   coreutils = all;
-  cpio = all;
```
Comment (teto) [inline pkgs/top-level/release-small.nix:36]: isn't that useful to deal/debug initramfs ? could be useful

**Review by teto — APPROVED** [review]

File: pkgs/top-level/release-small.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/top-level/release-small.nix:36]: It seemed to have a lot of overlap with `gnutar`. And `libarchive` seems to one-up it in almost every use case

Comment (teto) [reply pkgs/top-level/release-small.nix:36]: ok trust you with this :)

Comment (jonringer) [reply pkgs/top-level/release-small.nix:36]: These assumptions are based upon https://www.gnu.org/software/cpio/

I haven't actually heard of the tool before today.

**Comment by grahamc** [comment]
We definitely want to keep cpio, it is used for generating initrds. 

On February 2, 2022, GitHub ***@***.***> wrote:
> @jonringer commented on this pull request.
>

> In pkgs/top-level/release-small.nix
> <https://github.com/NixOS/nixpkgs/pull/157886#discussion_r798117707>:
>

> @@ -33,20 +33,17 @@ with import ./release-lib.nix { inherit
supportedSystems nixpkgsArgs; }; classpath = linux; cmake = all;
coreutils = all; - cpio = all;
> These assumptions are based upon <https://www.gnu.org/software/cpio/>
>
> I haven't actually heard of the tool before today.
>

—
Reply to this email directly, view it on GitHub
<https://github.com/NixOS/nixpkgs/pull/157886#discussion_r798117707>, or
unsubscribe <https://github.com/notifications/unsubscribe-
auth/AAASXLF7TI3D5JWEODLCJSTUZHCBNANCNFSM5NM4YZOA>.
Triage notifications on the go with GitHub Mobile for iOS
<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-
email&mt=8&pt=524675> or Android
<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-
email%26utm_medium%3Demail%26utm_source%3Dgithub>.
You are receiving this because you were mentioned.Message ID:
***@***.***>

**Comment by grahamc** [comment]
We should probably also keep mdadm. Otherwise this looks fairly good to
me.

**Comment by jonringer** [comment]
@grahamc applied! :)

**Review by roberth — APPROVED** [review]

**Pull request merged**